src/components/DatePicker.js

```diff
@@ -0,0 +1,454 @@
+const React = require('react');
+const Radium = require('radium');
+const moment = require('moment');
+
+const Icon = require('./Icon');
+
+class DatePicker extends React.Component {
+  constructor (props) {
+    super(props);
+    this.state = {
+      currentDate: null,
+      showCalendar: false
+    };
+  }
+
+  _shouldBorderCell (cellCount) {
+    return cellCount % this.props.calendarColumns !== 0;
+  }
+
+  _shouldBorderRow (currentCell, cellCount) {
+    const rows = (cellCount / this.props.calendarColumns) - 1;
+
+    return (currentCell / this.props.calendarColumns) < rows;
+  }
+
+  _handleDateSelect (date) {
+    if (this.props.closeOnDateSelect) {
+      this._handleBlur();
+    }
+
+    this.props.onDateSelect(date);
+  }
+
+  _handlePreviousClick () {
+    const selectedDate = moment.unix(this.props.selectedDate).locale(this.props.locale);
+    let currentDate = this.state.currentDate ? this.state.currentDate.locale(this.props.locale) : selectedDate;
+
+    currentDate = moment(currentDate.startOf('month').subtract(1, 'm'), this.props.format);
+
+    this.setState({
+      currentDate
+    });
```

Comment:
Should clicking the next/prev buttons actually change the selected date? I think a better UX would be to change the month, but not select a date. Leaving it up to the user to actually select the date they want.